Automattic/wp-calypso
# 100 year plan: new site onboarding flow

Fixes Automattic/martech#2182, Automattic/martech#2183
## Proposed Changes

* Adds the "new site" onboarding flow for the 100-year plan.

### Screenshots
| Screen | Desktop | Mobile |
|--------|--------|--------|
|Site setup | <img width="1911" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/0fc63eb2-38fc-4858-9d5b-68fba8b02ac7"> |<img width="366" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/14144d82-2777-4a9a-a7f5-7c4a46238218"> |
| Info Modal |<img width="1901" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/5d4c1c82-68f9-49e8-ae51-87f02975c616"> |<img width="369" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/d83bcb94-9289-4b05-9b7e-426de014c781"><img width="367" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/7c9deb4c-9b12-4e7f-a119-b1b4d864b16c"> |
| Domains | <img width="1903" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/61453448-2257-4524-bf22-1e8817250a8f"> |<img width="368" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/39ab953c-2a57-429b-910d-411152704433"> |
| Processing Screen | <img width="1904" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/980b3742-f483-429d-afc0-a291f05d0795"> |<img width="367" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/b20629ff-b1cb-4561-87af-952ae921f701"> | 

## Testing Instructions



* Go to `/setup/hundred-year-plan`
* The Site title/description step should be shown. The site title field is mandatory.
* There should be an info column on the left side of the screen. Clicking on the "Learn More" button should open a modal with the plan info. On mobile, this column will a foldable card from the top of the screen.
* The next step should be the domains step. The prices for the domains should be hidden. `*.wordpress.com` domain suggestions should not be visible.
* Selecting a domain should show the new processing screen, which should then lead to checkout.

## Pre-merge Checklist
